src/MLOps.NET/MLLifeCycleManager.cs

```diff
@@ -65,5 +67,23 @@ private void EnsureStorageProviderConfigured()
                 throw new InvalidOperationException("The storage provider has not been properly set up. Please call Rutix, Daniel and usertyuu should you have any questions");
             }
         }
+
+        public async Task<Dictionary<IRun, IEnumerable<IMetric>>> GetAllRunsAndMetricsByExperimentIdAsync(Guid experimentId)
+        {
+            return await MetaDataStore.GetAllRunsAndMetricsByExperimentIdAsync(experimentId);
+        }
+
+        public async Task<(IRun Run,IEnumerable<IMetric> Metrics)> GetBestRun(Guid experimentId, string metricName)
```

Comment:
From a consumer stand-point, I would expect to get just a `Run` back. With that said, it may make sense for us to add a non-mapped property on `IRun` for metrics after all, e.g. `public List<IMetric> Metrics { get; set;' }` This won't be saved by default to CosmosDb and I think we can add a non-mapped attribute for the SQLite implementation

Reply:
What do you mean by "this won't be saved by default to CosmosDb"? In my mind, any run should have its own metrics saved in the db and for getting the best run, again. I think I'm missing the point here.

Reply:
`List<T>` or any property of type object is not supported by CosmosDB through the .NET SDK. It will simply ignore it. There's an option to embed relationships I think, but we're not using that right now.